src/schema/posts.ts

```diff
@@ -881,12 +882,18 @@ export const resolvers: IResolvers<any, Context> = {
       ctx: Context,
       info,
     ): Promise<GQLPost> => {
-      const post = await ctx.con.getRepository(Post).findOneOrFail({
-        select: ['sourceId', 'private'],
+      const partialPost = await ctx.con.getRepository(Post).findOneOrFail({
+        select: ['id', 'sourceId', 'private'],
+        relations: ['source'],
         where: { id },
       });
-      if (post.private) {
-        await ensureSourcePermissions(ctx, post.sourceId);
+      const postSource = await partialPost.source;
```

Comment:
`source` is already preloaded with `relations` above, this is just to unwrap the `Promise`. 

Alternatively I could switch this to custom query to select just `source.type` if needed.

Reply:
That's fine